kratos/spatial_containers/tree.h

```diff
@@ -41,6 +42,17 @@ namespace Kratos
 ///@name  Functions
 ///@{
 
+// Helper template to extract ObjectType or default to void
+template <typename T, typename = void>
+struct GetObjectType {
```

Comment:
I think this is too generic to be in the global namespace. Shall we protect it with Internals namespace or give it a Tree prefix to have it tied to it?

Reply:
Okay

Reply:
https://github.com/KratosMultiphysics/Kratos/pull/11739 here